### PR TITLE
Also search the path for executable

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: src/filesystem.cpp [hunk 1]
```diff
@@ -1002,18 +1004,34 @@ std::string get_exe_dir()
...
-		return get_cwd();
+	// check the PATH for wesnoth's location
+	// versionless
+	std::string exe = filesystem::base_name(filesystem::get_program_invocation("wesnoth"));
+	bfs::path search = bp::search_path(exe).string();
+	if(!search.string().empty()) {
+		return search.parent_path().string();
+	}
+
+	// with version
```
Comment (soliton-) [inline src/filesystem.cpp:1025]: I'd say with version should be tried first then.

File: src/filesystem.cpp [hunk 2]
```diff
@@ -1002,18 +1004,34 @@ std::string get_exe_dir()
...
+			return exe.parent_path().string();
 		}
+	}
 
-		return exe.parent_path().string();
-	} else {
-		return get_cwd();
+	// check the PATH for wesnoth's location
+	// versionless
+	std::string exe = filesystem::base_name(filesystem::get_program_invocation("wesnoth"));
```
Comment (soliton-) [inline src/filesystem.cpp:1019]: It's never a path now, right? So no point in calling base_name.

File: src/filesystem.cpp [hunk 3]
```diff
@@ -1698,7 +1716,7 @@ std::string get_program_invocation(const std::string& program_name)
 #endif
 	);
 
-	return (bfs::path(game_config::wesnoth_program_dir) / real_program_name).string();
```
Comment (soliton-) [inline src/filesystem.cpp:1701]: `game_config::wesnoth_program_dir` can be completely removed now. It's not used anywhere else.

File: src/filesystem.cpp (hunk 1 above)
Comment (Pentarctagon) [reply src/filesystem.cpp]: Updated.

File: src/filesystem.cpp (hunk 2 above)
Comment (Pentarctagon) [reply src/filesystem.cpp]: Updated.

File: src/filesystem.cpp (hunk 3 above)
Comment (Pentarctagon) [reply src/filesystem.cpp:1701]: Updated.

**Review by soliton- — APPROVED** [review]

**Pull request merged**